amberdata/src/endpoint/balance.ts

```diff
@@ -0,0 +1,114 @@
+import objectPath from 'object-path'
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { AdapterRequest, Config, Address, Account } from '@chainlink/types'
+import { DEFAULT_DATA_PATH, getBaseURL } from '../config'
+import { BLOCKCHAINS, isCoinType, isChainType } from '.'
+
+export const Name = 'balance'
+
+type RequestData = {
+  dataPath: string
+  confirmations: number
+}
+
+type ResponseWithResult = {
+  response: any
+  result: Account
+  warning?: string
+}
+
+const WARNING_NO_OPERATION_COIN = 'No Operation: unsupported coin'
+const WARNING_NO_OPERATION_TESTNET = "No Operation: 'testnet' is only supported on 'eth'"
+const WARNING_NO_OPERATION_CHAIN =
+  "No Operation: invalid chain parameter, must be one of 'mainnet' or 'testnet'"
+const WARNING_NO_OPERATION_MISSING_ADDRESS = 'No Operation: address param is missing'
+
+const getBalanceURI = (address: string) => `/api/v2/addresses/${address}/account-balances/latest`
+
+const getBlockchainHeader = (chain: string, coin: string) => {
+  const network = Requester.toVendorName(coin, BLOCKCHAINS)
+  if (chain === 'testnet') {
+    if (network === 'ethereum') return 'ethereum-rinkeby'
+    throw WARNING_NO_OPERATION_TESTNET
+  }
+  return `${network}-mainnet`
+}
+
+const getBalances = async (config: Config, addresses: Address[]): Promise<ResponseWithResult[]> =>
+  Promise.all(
+    addresses.map(async (addr: Address) => {
+      const warnedResponse = {
+        result: { ...addr, balance: 0 },
+        response: null,
+      }
+
+      if (!addr.address) return { ...warnedResponse, warning: WARNING_NO_OPERATION_MISSING_ADDRESS }
+
+      if (!addr.coin) addr.coin = 'btc'
+      if (isCoinType(addr.coin) === false)
+        return {
+          ...warnedResponse,
+          warning: WARNING_NO_OPERATION_COIN,
+        }
+
+      if (!addr.chain) addr.chain = 'mainnet'
+      if (isChainType(addr.chain) === false)
```

Comment:
```suggestion
      if (!isChainType(addr.chain))
```

Reply:
This seems like a style choice of clear vs. readable, but if you have a preference I'm fine with switching. I'd agree that the not operator is easier to read.

I read some opinion piece a while ago that argued being explicit is better, especially for helping the understanding of newer devs.